File: plugins/devkit/devkit-java-tests/testData/inspections/unspecifiedActionsPlace/UnspecifiedActionPlaces.java
import com.intellij.openapi.actionSystem.*;
public class UnspecifiedActionPlaces {

  private final static String CUSTOM_UNKNOWN = "unknown";
  private final static String CUSTOM_UNKNOWN_FROM_PLATFORM_VALUE = ActionPlaces.UNKNOWN;

  public void foo() {
    ActionManager actionManager = ActionManager.getInstance();
    // popup menu:
    actionManager.createActionPopupMenu("asdasd", new ActionGroup());
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">""</warning>, new ActionGroup());
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">"unknown"</warning>, new ActionGroup());
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">ActionPlaces.UNKNOWN</warning>, new ActionGroup());
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">CUSTOM_UNKNOWN</warning>, new ActionGroup());
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">CUSTOM_UNKNOWN_FROM_PLATFORM_VALUE</warning>, new ActionGroup());
    // toolbar:
    actionManager.createActionToolbar("asdasd", new ActionGroup(), false);
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">""</warning>, new ActionGroup(), false);
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">"unknown"</warning>, new ActionGroup(), false);
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">ActionPlaces.UNKNOWN</warning>, new ActionGroup(), false);
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">CUSTOM_UNKNOWN</warning>, new ActionGroup(), false);
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">CUSTOM_UNKNOWN_FROM_PLATFORM_VALUE</warning>, new ActionGroup(), false);
  }
}

File: plugins/devkit/devkit-kotlin-tests/testData/inspections/unspecifiedActionsPlace/UnspecifiedActionPlaces.kt
import com.intellij.openapi.actionSystem.*

class UnspecifiedActionPlaces {

  companion object {
    private const val CUSTOM_UNKNOWN = "unknown"
    private const val CUSTOM_UNKNOWN_FROM_PLATFORM_VALUE = ActionPlaces.UNKNOWN
  }

  fun any() {
    val actionManager = ActionManager.getInstance()
    // popup menu:
    actionManager.createActionPopupMenu("asdasd", ActionGroup())
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">""</warning>, ActionGroup())
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">"unknown"</warning>, ActionGroup())
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">ActionPlaces.UNKNOWN</warning>, ActionGroup())
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">CUSTOM_UNKNOWN</warning>, ActionGroup())
    actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">CUSTOM_UNKNOWN_FROM_PLATFORM_VALUE</warning>, ActionGroup())
    // toolbar:
    actionManager.createActionToolbar("asdasd", ActionGroup(), false)
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">""</warning>, ActionGroup(), false)
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">"unknown"</warning>, ActionGroup(), false)
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">ActionPlaces.UNKNOWN</warning>, ActionGroup(), false)
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">CUSTOM_UNKNOWN</warning>, ActionGroup(), false)
    actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">CUSTOM_UNKNOWN_FROM_PLATFORM_VALUE</warning>, ActionGroup(), false)
  }
}

// test in top-level function:
fun topLevelFunction() {
  val actionManager = ActionManager.getInstance()
  // popup menu:
  actionManager.createActionPopupMenu("asdasd", ActionGroup())
  actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">""</warning>, ActionGroup())
  actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">"unknown"</warning>, ActionGroup())
  actionManager.createActionPopupMenu(<warning descr="Unspecified place for action popup menu">ActionPlaces.UNKNOWN</warning>, ActionGroup())
  // toolbar:
  actionManager.createActionToolbar("asdasd", ActionGroup(), false)
  actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">""</warning>, ActionGroup(), false)
  actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">"unknown"</warning>, ActionGroup(), false)
  actionManager.createActionToolbar(<warning descr="Unspecified place for action toolbar">ActionPlaces.UNKNOWN</warning>, ActionGroup(), false)
}
